app/Services/UserService.php

```diff
@@ -15,4 +18,36 @@ public function createUser(array $data)
         $user_data['password'] = Hash::make($data['password']);
         return User::create($user_data);
     }
+
+    public function resetPassword(array $data, string $token)
+    {
+        $userID = DB::table('users')->where('email', $data['email'])->value('id');
+        if ($userID != null) {
+            ResetPassword::updateOrCreate(
+                ['user_id' => $userID],
+                [
+                    'user_id' => $userID,
+                    'token' => $token
+                ]
+            );
+        } else {
+            return response()->json(['User not found!', 404]);
+        }
+    }
+    public function setNewPassword(array $data)
+    {
+        $token = $data['token'];
+        $password = Hash::make($data['password']);
+        $userID = DB::table('reset_passwords')->where('token', $token)->value('user_id');
+        if ($userID != null) {
+            SetNewPassword::updateOrCreate(
```

Comment:
what is the 'SetNewPassword' ? we need to save a new password to the user model. $user->password = $newPassword; $user->save();